### PR TITLE
[Doc] Add documentation for the Menu component

### PR DESCRIPTION
**Pull request by fzaninotto**
image

### CODE REVIEW

File: docs/Layout.md [hunk 1]
```diff
@@ -47,17 +50,25 @@ const App = () => (
 | `appBar`    | Optional | `Component` | -        | A React component rendered at the top of the layout                   |
 | `className` | Optional | `string`    | -        | Passed to the root `<div>` component                                  |
 | `error`     | Optional | `Component` | -        | A React component rendered in the content area in case of error       |
-| `sidebar`   | Optional | `Component` | -        | A React component rendered at the side of the screen                  |
```
Comment (fzaninotto) [inline docs/Layout.md:50]: Even if the `sidebar` prop exists, I see no reason to use it, so let's make it a private API. `menu`, on the other side, is the preferred way to set a custom menu, so it should be documented.

File: packages/ra-ui-materialui/src/layout/Menu.tsx [hunk 2]
```diff
@@ -16,16 +17,37 @@ import { useSidebarState } from './useSidebarState';
...
+ *     </Menu>
+ * );
+ */
 export const Menu = (props: MenuProps) => {
     const resources = useResourceDefinitions();
     const getResourceLabel = useGetResourceLabel();
     const createPath = useCreatePath();
     const {
         hasDashboard,
-        dense,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/Menu.tsx:25]: passed as rest props to the root component, a mui `MenuList`

File: packages/ra-ui-materialui/src/layout/Menu.tsx [hunk 3]
```diff
@@ -84,14 +105,18 @@ Menu.propTypes = {
     hasDashboard: PropTypes.bool,
 };
 
+// re-export MenuItem commponents for convenience
+Menu.Item = MenuItemLink;
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/Menu.tsx:109]: This is technically a new feature but I think it's really minor, plus it's backwards compatible.

File: packages/ra-ui-materialui/src/layout/MenuItemLink.tsx [hunk 4]
```diff
@@ -137,17 +137,16 @@ export const MenuItemLink = forwardRef((props: MenuItemLinkProps, ref) => {
     );
 });
 
-interface Props {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/MenuItemLink.tsx:140]: This change is required to let TS compile the Menu component after it reexports the MenuItem component.

**Review by septentrion-730n — APPROVED** [review]

File: docs/Admin.md [hunk 5]
```diff
@@ -313,51 +313,40 @@ React-admin uses the list of `<Resource>` components passed as children of `<Adm
...
 
 Then, pass it to the `<Admin>` component as the `menu` prop:
 
 ```jsx
 // in src/App.js
-import Menu from './Menu';
+import { MyMenu } from './Menu';
 
 const App = () => (
     <Admin menu={Menu} dataProvider={simpleRestProvider('http://path.to.my.api')}>
```
Comment (slax57) [inline docs/Admin.md:352]:
```suggestion
    <Admin menu={MyMenu} dataProvider={simpleRestProvider('http://path.to.my.api')}>
```

File: docs/Layout.md [hunk 6]
```diff
@@ -197,23 +208,47 @@ export const MyError = ({
...
+export const Layout = (props) => <Layout {...props} menu={MyMenu} />;
+```
 
-export default MyLayout;
+You can create a custom menu component using [react-admin's `<Menu>` component](./Menu.md):
+
+```jsx
+// in src/MyMenu.js
+import * as React from 'react';
+import { DashboardMenuItem, Menu, MenuItemLink } from 'react-admin';
```
Comment (slax57) [inline docs/Layout.md:229]:
```suggestion
import { Menu } from 'react-admin';
```

File: docs/Layout.md [hunk 7]
```diff
@@ -197,23 +208,47 @@ export const MyError = ({
...
+        <Menu.DashboardItem />
+        <Menu.Item to="/posts" primaryText="Posts" leftIcon={<BookIcon />}/>
+        <Menu.Item to="/comments" primaryText="Comments" leftIcon={<ChatBubbleIcon />}/>
+        <Menu.Item to="/users" primaryText="Users" leftIcon={<PeopleIcon />}/>
+        <Menu.Item to="/custom-route" primaryText="Miscellaneous" leftIcon={<LabelIcon />}/>
+    </Menu>
+);
 ```
 
+But you don't have to use react-admin-s `<Menu>` component. In fact, the `<Layout menu>` component can render anything you like, e.g. using [MUI's `<Menu>` component](https://mui.com/material-ui/react-menu/).
```
Comment (slax57) [inline docs/Layout.md:246]:
```suggestion
But you don't have to use react-admin's `<Menu>` component. In fact, the `<Layout menu>` component can render anything you like, e.g. using [MUI's `<Menu>` component](https://mui.com/material-ui/react-menu/).
```

File: docs/Menu.md [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+
+# `<Menu>`
+
+This component renders a menu, with one menu item per resource by default. You can also set menu items by hand.
+
+![standard menu](./img/menu.webp)
+
+## Usage
+
+You can create a custom menu component using react-admin's `<Menu>` and `<MenuItemLink>` components:
```
Comment (slax57) [inline docs/Menu.md:14]:
```suggestion
You can create a custom menu component using react-admin's `<Menu>` and `<Menu.Item>` components:
```

File: docs/Menu.md [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+        <Menu.DashboardItem />
+        <Menu.Item to="/posts" primaryText="Posts" leftIcon={<BookIcon />}/>
+        <Menu.Item to="/comments" primaryText="Comments" leftIcon={<ChatBubbleIcon />}/>
+        <Menu.Item to="/users" primaryText="Users" leftIcon={<PeopleIcon />}/>
+        <Menu.Item to="/custom-route" primaryText="Miscellaneous" leftIcon={<LabelIcon />}/>
+    </Menu>
+);
+```
+
+Chech [the `<Menu.Item>` section](#menuitem) for more information.
```
Comment (slax57) [inline docs/Menu.md:120]:
```suggestion
Check [the `<Menu.Item>` section](#menuitem) for more information.
```

File: packages/ra-ui-materialui/src/layout/Menu.tsx [hunk 10]
```diff
@@ -84,14 +105,18 @@ Menu.propTypes = {
...
+
 const PREFIX = 'RaMenu';
 
 export const MenuClasses = {
     open: `${PREFIX}-open`,
     closed: `${PREFIX}-closed`,
 };
 
-const Root = styled('div', {
+const Root = styled(MenuList, {
```
Comment (slax57) [inline packages/ra-ui-materialui/src/layout/Menu.tsx:119]: Isn't this a BC (for styling)?

**Review by slax57 — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/layout/Menu.tsx (hunk 10 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/layout/Menu.tsx:119]: it's not: the style is the same

**Pull request merged**